src/flb_processor.c

```diff
@@ -566,6 +567,15 @@ static int load_from_config_format_group(struct flb_processor *proc, int type, s
             if (pair->val->type != CFL_VARIANT_STRING) {
                 continue;
             }
+            /* If filter plugin in processor unit has its own match rule,
+             * we must release the pre-allocated '*' match at first.
+             */
+            if (pu->unit_type == FLB_PROCESSOR_UNIT_FILTER) {
+                if (strcmp(pair->key, "match") == 0) {
+                    f_ins = (struct flb_filter_instance *)pu->ctx;
+                    flb_sds_destroy(f_ins->match);
```

Comment:
Just to be sure there are no issues in the future wrap this `flb_sds_destroy` with a conditional that ensures that `f_ins->match` is not `NULL` and afterwards set `f_ins->match` to `NULL`.

Even though the call to `flb_processor_unit_set_property` SHOULD set `match` appropriately, if we don't set it to `NULL` here and that call fails the cleanup process will try to release an sds instance that was already disposed of.